internal/schemaver/string.go

```diff
@@ -0,0 +1,95 @@
+package schemaver
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+type String string
+
+func NewString(model, revision, addition int) String {
+	return String(fmt.Sprintf("%d.%d.%d", model, revision, addition))
+}
+
+func ParseAsString(s string) (String, error) {
+	// must provide model.revision.addition
+	cleaned := strings.TrimSpace(s)
+	parts := strings.Split(cleaned, ".")
+	if len(parts) != 3 {
+		return "", fmt.Errorf("invalid schema version format: %s", s)
+	}
+	// check that all parts are integers
+	for _, part := range parts {
+		v, err := strconv.Atoi(part)
+		if err != nil || v < 0 {
+			return "", fmt.Errorf("invalid schema version format: %s", s)
+		}
+	}
+	return String(cleaned), nil
+}
+
+func (s String) String() string {
+	return string(s)
+}
+
+func (s String) ModelPart() (int, bool) {
+	v, ok := parseVersionPart(s, 0)
+	if v == 0 {
+		ok = false
+	}
+	return v, ok
+}
+
+func (s String) RevisionPart() (int, bool) {
+	return parseVersionPart(s, 1)
+}
+
+func (s String) AdditionPart() (int, bool) {
+	return parseVersionPart(s, 2)
+}
+
+func (s String) LessThan(other String) bool {
+	return s.compare(other) < 0
+}
+
+func (s String) GreaterOrEqualTo(other String) bool {
+	return s.compare(other) >= 0
+}
+
+func (s String) compare(other String) int {
```

Comment:
I'm sort of surprised that we have this logic on both the String representation and Struct representation of the SchemaVersion. Can you help me understand why?

Reply:
Originally this was used within structs representing JSON that were marshalled to strings. I think at this point I could transition that to only the struct representation (the string approach might be an unnecessary holdover). Let me get back to you.

Reply:
I've replaced all usages of the string variant with the struct version